cmd/client-s3-trace_v2.go

```diff
@@ -66,23 +65,7 @@ func (t traceV2) Response(resp *http.Response) (err error) {
 		resp.StatusCode != http.StatusNoContent {
 		respTrace, err = httputil.DumpResponse(resp, true)
 	} else {
-		// WORKAROUND for https://github.com/golang/go/issues/13942.
-		// httputil.DumpResponse does not print response headers for
-		// all successful calls which have response ContentLength set
-		// to zero. Keep this workaround until the above bug is fixed.
-		if resp.ContentLength == 0 {
-			var buffer bytes.Buffer
-			if err = resp.Header.Write(&buffer); err != nil {
-				return err
-			}
-			respTrace = buffer.Bytes()
-			respTrace = append(respTrace, []byte("\r\n")...)
-		} else {
-			respTrace, err = httputil.DumpResponse(resp, false)
-			if err != nil {
-				return err
-			}
-		}
+		respTrace, err = httputil.DumpResponse(resp, false)
```

Comment:
Why are we not returning on error here? If we don't return this code will get executed,

```if globalInsecure && resp.TLS != nil {
       dumpTLSCertificates(resp.TLS)
    }
```

Is that intended?

Reply:
it really doesn't matter..

Reply:
ok